### PR TITLE
completes Q5 get-all-articles

### PR DESCRIPTION
**Pull request by robbochobbo**
Adds /api/articles endpoint

### CODE REVIEW

**Comment by robbochobbo** [comment]
I meant to delete the req.query part of the getAllArticles function in the controller!

**Comment by robbochobbo** [comment]
I meant to delete the req.query variables that I pass into getAllArticles in my controller... tried to think too far ahead there!

File: __tests__/app.test.js [hunk 1]
```diff
@@ -146,6 +147,35 @@ describe('/api/articles/:article_id', () => {
     })
 })
 
+describe('/api/articles', () => {
+    test('GET 200: responds with an array of objects of articles, each with the correct properties', () => {
+        return request(app)
+        .get('/api/articles')
+        .expect(200)
+        .then(({body: {articles}}) => {
+            articles.forEach((article) => {
```
Comment (nc-saima-khanom) [inline __tests__/app.test.js:156]: As we are using a forEach here - if the array is empty then this test will pass by default as none of the expect statements will be called. A test with no expect statements will pass by default. We should first check that there is something in the array (check its length)

File: __tests__/app.test.js [hunk 2]
```diff
@@ -146,6 +147,35 @@ describe('/api/articles/:article_id', () => {
...
+            })
+        })
+    })
+
+    test('GET 200: responds with articles sorted by date descending', () => {
+        return request(app)
+        .get('/api/articles')
+        .expect(200)
+        .then(({body: {articles}}) => {
+            expect(articles).toBeSortedBy('created_at',{descending:true})
```
Comment (nc-saima-khanom) [inline __tests__/app.test.js:174]: nice!

File: __tests__/app.test.js [hunk 3]
```diff
@@ -146,6 +147,35 @@ describe('/api/articles/:article_id', () => {
...
+        .then(({body: {articles}}) => {
+            articles.forEach((article) => {
+                expect(typeof article.author).toBe('string')
+                expect(typeof article.title).toBe('string')
+                expect(typeof article.article_id).toBe('number')
+                expect(typeof article.topic).toBe('string')
+                expect(typeof article.created_at).toBe('string')
+                expect(typeof article.votes).toBe('number')
+                expect(typeof article.article_img_url).toBe('string')
+                expect(typeof article.comment_count).toBe('string')
```
Comment (nc-saima-khanom) [inline __tests__/app.test.js:164]: we're also expecting it to not have a body property!

File: models/models.js [hunk 4]
```diff
@@ -60,7 +50,30 @@ const fetchArticleById = (article_id) => {
     });
 }
 
+const fetchAllArticles = (sort_by='created_at',  order='DESC') => {
+
+    return db
+    .query(`SELECT
```
Comment (nc-saima-khanom) [inline models/models.js:56]: nice!

File: __tests__/app.test.js (hunk 3 above)
Comment (robbochobbo) [reply __tests__/app.test.js:164]: would that just be expect(typeof article.body).toBe('undefined') ?

**Pull request merged**